core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -251,7 +277,19 @@ public ServerBuilder port(InetSocketAddress localAddress, SessionProtocol protoc
      * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
      */
     public ServerBuilder port(ServerPort port) {
```

Comment:
How common do you think it is to have `HTTP` and `HTTPS` on the same port? I sort of want to log something when both are enabled since doing it accidentally could be a major security hole.

Reply:
What do you think about adding a method `usePortUnification()` and allowing only when it's called in prior?

Reply:
Added `allowPortUnification()`:

```java
ServerBuilder sb = new ServerBuilder();
sb.allowPortUnification(PROXY, HTTPS);
sb.proxyProtocol(443); // OK
sb.https(443); // OK
sb.http(443); // Not OK
```

@minwoox @anuraaga WDYT?

Reply:
Actually PROXY has to be allowed even without allowPortUnification. Let me update the PR.

Reply:
Updated:

```java
ServerBuilder sb = new ServerBuilder();
sb.proxyProtocol(443); // OK
sb.https(443); // OK
sb.http(443); // Not OK

ServerBuilder sb2 = new ServerBuilder();
sb2.usePortUnification();
sb2.proxyProtocol(443); // OK
sb2.https(443); // OK
sb2.http(443); // OK
```

Reply:
The current API doesn't fit very well when it comes to ephemeral port (0):

```java
sb.proxyProtocol(0);
sb.http(0);
sb.https(0);
``

The code above gives us various questions:

- Should `http(0)` and `https(0)` be bound to the same port?
- Where should `proxyProtocol(0)` be applied, to HTTP, HTTPS or both?

Let me:

- Remove `usePortUnification()`,
- Remove `proxyProtocol()` and
- Add `port(int, SessionProtocol...)` and its variants which would be the only way to enable PROXY protocol and/or port unification.